lib/testing/src/lib/core/utils/browser-actions.ts

```diff
@@ -80,6 +89,16 @@ export class BrowserActions {
         await elementFinder.sendKeys(text);
     }
 
+    static async sendKeysIfVisible(elementFinder: ElementFinder, text: string) {
```

Comment:
All seems ok to be the only thing I am not sure if its a good idea to have methods like that or not. @gmandakini @adinapitul feelings about it?

Reply:
TBH, I don't understand the need for these 2 sendKeysIf.. methods. What problem are they solving?

what is your concern about these, @eromano ?

Reply:
what is the difference? Having the same kind of method for visible and present...

Reply:
These 2 methods add an extra check for the element to be visible or present, before sending the keys.
The difference between the 2 methods is that the element can be present in the DOM, but not visible, still we need to be able to sendkeys to it.

Reply:
can _clearSendKeys_ be used instead of sendKeysIfVisible ?

in any case, naming wise I think I would use "when" instead of "if" . English wise, _when_ implies waiting. "if" makes me wonder what if it's not? :)

Reply:
Since the 2 methods generated to much discussion, I will remove them.

Reply:
Like more of the methods from this file, the 2 I added are using a default Webdriver method (sendkeys) with an additional explicit wait for the element to be visible or present.
It would be great to at least understand what is causing so much confusion/discussions around them.
I'm removing both just because they are not in scope of this PR, but I've used them in my previous tests and I found them useful.